optuna/testing/storages.py

```diff
@@ -114,3 +114,14 @@ def __exit__(
             self.thread.join()
             self.server = None
             self.thread = None
+
+
+def _find_free_port() -> int:
+    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    for port in range(13000, 13100):
+        try:
+            sock.bind(("localhost", port))
+            return port
+        except OSError:
+            continue
+    assert False, "must not reach here"
```

Comment:
I'm afraid `assert False` gets ignored if user set `PYTHONOPTIMIZE` environment variable.

Isn't it better to explicitly raise an exception?

```suggestion
    raise RuntimeError("No free port available")
```

Reply:
@ktns Thank you for your feedback! The `optuna.testing` package is introduced for unit tests. I would say the use of `assert` statements is acceptable in this context. What are your thoughts on this?

Reply:
My concern is that user might run unit tests accidentally with the `PYTHONOPTIMIZE` and an error message will be very confusing. The possibility might be slim but I would like to eliminate any possible confusion if there's no strong reason to use `assert False`.

Reply:
Optuna uses `pytest` as a test runner, and the unit tests rely on `assert` statements.

@HideakiImamura I'll ask for your input and let you make the final decision. Thank you for bringing this up.

Reply:
Given that the developers are the ones who execute the test, it seems natural to require reading the code to interpret the results. Also, the current practice of using 'assert False' is common in Optuna. Therefore, how about we proceed with the merge as is?